### PR TITLE
changing to zap log

### PR DESCRIPTION
**Pull request by nithu0115**
Issue #, if available: N/A

Description of changes:

Switching to zap core logging instead of `klog`. Part of the motivation for `logr` was the idea that V level is a bad idea - log lines are all INFO or ERROR. But there are real world scenario when we want know more on what is happening and it is required. We should limit to V(1) (Debug logging) and should not go over that level. 

Changes:
1) Added log-level arg which will either take `debug` or `info` log levels 
2) Modified manager.go template to pass in region value to generate create temp credentials to be consistent with config region flag

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by nithu0115** [comment]
(Note to reviewer: Change to templates/pkg/crd_manager.go.tpl will require us to re-generate Service controllers)

**Review by A-Hilaly — APPROVED** [review]
/lgtm 👍 

Shall we regenerate the service controllers in this PR ?

File: pkg/runtime/config.go [hunk 1]
```diff
@@ -28,6 +29,7 @@ const (
 	flagEnableDevLogging     = "enable-development-logging"
 	flagAWSAccountID         = "aws-account-id"
 	flagAWSRegion            = "aws-region"
+	flagLogLevel			 = "log-level"
```
Comment (A-Hilaly) [inline pkg/runtime/config.go:32]: nit: please run go fmt :)

File: pkg/runtime/session.go [hunk 2]
```diff
@@ -13,12 +13,16 @@
 
 package runtime
 
-import "github.com/aws/aws-sdk-go/aws/session"
+import (
+	ackv1alpha1 "github.com/aws/aws-controllers-k8s/apis/core/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
```
Comment (A-Hilaly) [inline pkg/runtime/session.go:19]: nit: let's keep internal and external imports separated

File: pkg/runtime/reconciler.go [hunk 3]
```diff
@@ -15,7 +15,6 @@ package runtime
 
 import (
 	"context"
-
```
Comment (A-Hilaly) [inline pkg/runtime/reconciler.go:18]: the new line was used to keep standard imports and external imports separated

**Review by A-Hilaly — COMMENTED** [review]
leaving few comments inline

**Comment by nithu0115** [comment]
> Shall we regenerate the service controllers in this PR ?

I am happy to re-generate but wanted to check in with @jaypipes to see if we want to introduce any process around this. Fwiw, we should maintain 2 branches (main and dev/staging). This way, whenever we make such breaking changes, we can push to dev and cherry pick to merge into main once we cut new branches.

**Comment by nithu0115** [comment]
> > Shall we regenerate the service controllers in this PR ?
> 
> I am happy to re-generate but wanted to check in with @jaypipes to see if we want to introduce any process around this. Fwiw, we should maintain 2 branches (main and dev/staging). This way, whenever we make such breaking changes, we can push to dev and cherry pick to merge into main once we cut new releases.

Spoke to @jaypipes offline and come to a consensus to re-generate service controllers as this is a minimal change. If the change is breaking one or a big one, we will go ahead and merge the PR first without re-generating service controller(s) and then open new PRs for each controller.

File: pkg/runtime/reconciler.go [hunk 4]
```diff
@@ -98,11 +98,12 @@ func (r *reconciler) reconcile(req ctrlrt.Request) error {
 	acctID := r.getOwnerAccountID(res)
 	region := r.getRegion(res)
 
-	r.log.WithValues(
+	log := r.log.WithValues(
 		"account_id", acctID,
 		"region", region,
 		"kind", r.rd.GroupKind().String(),
 	)
+	log.Info("starting reconcilation")
```
Comment (jaypipes) [inline pkg/runtime/reconciler.go:106]: This should be DEBUG, not INFO :)

File: pkg/runtime/session.go [hunk 5]
```diff
@@ -13,12 +13,17 @@
...
-func NewSession() (*session.Session, error) {
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func NewSession(region ackv1alpha1.AWSRegion) (*session.Session, error) {
 	// NOTE(jaypipes): session.NewSession() is needed for the STS::AssumeRole
 	// stuff we will need to do...
-	sess, err := session.NewSession()
+	sess, err := session.NewSession(aws.NewConfig().WithRegion(string(region)))
```
Comment (jaypipes) [inline pkg/runtime/session.go:26]: This change is not related to logging. Would you mind removing this change (put it into a separate PR please)

File: services/bookstore/pkg/resource/book/manager.go [hunk 6]
```diff
@@ -150,7 +150,7 @@ func newResourceManager(
 	id ackv1alpha1.AWSAccountID,
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
Comment (jaypipes) [inline services/bookstore/pkg/resource/book/manager.go:153]: ditto.

File: services/petstore/pkg/resource/pet/manager.go [hunk 7]
```diff
@@ -227,7 +227,7 @@ func newResourceManager(
 	id ackv1alpha1.AWSAccountID,
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
Comment (jaypipes) [inline services/petstore/pkg/resource/pet/manager.go:230]: ditto

File: services/ecr/pkg/resource/repository/manager.go [hunk 8]
```diff
@@ -153,7 +153,7 @@ func newResourceManager(
 	id ackv1alpha1.AWSAccountID,
 	region ackv1alpha1.AWSRegion,
 ) (*resourceManager, error) {
-	sess, err := ackrt.NewSession()
+	sess, err := ackrt.NewSession(region)
```
Comment (jaypipes) [inline services/ecr/pkg/resource/repository/manager.go:156]: ditto.

File: services/sns/config/controller/deployment.yaml [hunk 9]
```diff
@@ -27,11 +27,11 @@ spec:
         - ./bin/controller
         args:
         - --aws-account-id
-        - "$(AWS_ACCOUNT_ID)"
+        - "${AWS_ACCOUNT_ID}"
         - --aws-region
-        - "$(AWS_REGION)"
+        - "${AWS_REGION}"
         - --enable-development-logging
-        - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
```
Comment (jaypipes) [inline services/sns/config/controller/deployment.yaml:34]: These changes need to be undone. The parentheses are necessary. The curly braces are incorrect for Docker args.

File: templates/config/controller/deployment.yaml.tpl [hunk 10]
```diff
@@ -27,11 +27,11 @@ spec:
...
         args:
         - --aws-account-id
-        - "$(AWS_ACCOUNT_ID)"
+        - "${AWS_ACCOUNT_ID}"
         - --aws-region
-        - "$(AWS_REGION)"
+        - "${AWS_REGION}"
         - --enable-development-logging
-        - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
+        - "${ACK_ENABLE_DEVELOPMENT_LOGGING}"
```
Comment (jaypipes) [inline templates/config/controller/deployment.yaml.tpl:34]: These changes need to be undone. The parentheses are necessary. The curly braces are incorrect for Docker args.

**Review by jaypipes — CHANGES_REQUESTED** [review]
@nithu0115 this needs to be redone. I support the changes for logging, but the remaining changes either belong in a separate PR (NewSession(region)) or are incorrect (the curly braces instead of parens)

File: pkg/runtime/reconciler.go (hunk 4 above)
Comment (nithu0115) [reply pkg/runtime/reconciler.go]: Done :)

File: templates/config/controller/deployment.yaml.tpl (hunk 10 above)
Comment (nithu0115) [reply templates/config/controller/deployment.yaml.tpl]: I wanted to keep this consistent with what we currently have - https://github.com/aws/aws-controllers-k8s/blob/main/services/ecr/config/controller/deployment.yaml#L34. Thus, I made this change.

**Comment by nithu0115** [comment]
> @nithu0115 this needs to be redone. I support the changes for logging, but the remaining changes either belong in a separate PR (NewSession(region)) or are incorrect (the curly braces instead of parens)

Addressed your comments.

**Review by jaypipes — APPROVED** [review]
:+1:
